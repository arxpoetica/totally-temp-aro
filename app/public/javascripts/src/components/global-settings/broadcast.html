<div class="no-collapse" style="display: flex; flex-direction: column; height: 100%;">
  <div style="flex: 1 1 auto;">
    <div class="form-group">
      <label>Subject</label>
      <input type="text" class="form-control" ng-model="$ctrl.mailSubject">
    </div>
    <div class="form-group">
      <label>Text</label>
      <textarea class="form-control" rows="10" ng-model="$ctrl.mailBody"></textarea>
    </div>
  </div>
  <div style="flex: 0 0 auto;">
    <button class="btn btn-primary float-right" ng-click="$ctrl.send()"><i class="fa fa-save"></i>&nbsp;&nbsp;Send</button>
  </div>
</div>