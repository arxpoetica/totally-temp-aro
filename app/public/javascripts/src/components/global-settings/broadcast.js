class BroadcastController {
  constructor ($http, $timeout, state, globalSettingsService) {
    this.state = state
    this.globalSettingsService = globalSettingsService
    this.$http = $http
    this.$timeout = $timeout

    this.mailSubject = null
    this.mailBody = null
  }

  send() {
    this.$http.post('/socket/broadcast', {
      subject: this.mailSubject,
      body: this.mailBody
    })
    .catch((err) => console.error(err))
  }
}

BroadcastController.$inject = ['$http', '$timeout', 'state', 'globalSettingsService']

let broadcast = {
  templateUrl: '/components/global-settings/broadcast.html',
  controller: BroadcastController
}

export default broadcast
