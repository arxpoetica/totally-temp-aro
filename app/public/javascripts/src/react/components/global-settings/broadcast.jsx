import React, { Component } from 'react'
import { connect } from 'react-redux'
import UiActions from '../configuration/ui/ui-actions'

export class Broadcast extends Component {
  constructor(props) {
    super(props)
    const { subject, message, startDate, endDate } = { ...this.props.broadcast }
    this.state = {
      subject: subject || '',
      message: message || '',
      startDate: startDate || '',
      endDate: endDate || '',
    }
  }

  render() {
    const { subject, message, startDate, endDate } = this.state
    return (
      <div className="no-collapse" style={{ display: 'flex', flexDirection: 'column', height: '100%' }}>
        <div style={{ flex: '1 1 auto' }}>
          <div className="form-group">
            <label className="font-weight-bold">Subject</label>
            <input
              name="subject"
              type="text"
              className="form-control"
              value={subject}
              onChange={(event) => this.handleOnChange(event)}
            />
          </div>
          <div className="form-group">
            <label className="font-weight-bold">Message</label>
            <textarea
              name="message"
              className="form-control"
              rows="12"
              value={message}
              onChange={(event) => this.handleOnChange(event)}
            />
          </div>
          <div className="form-group">
            <label className="font-weight-bold">When should the messages broadcast?</label>
            <div className="row">
              <div className="col-md-5">
                <input
                  type="date"
                  name="startDate"
                  className="form-control"
                  value={startDate}
                  onChange={(event) => this.handleOnChange(event)}
                />
              </div>
              <div className="col-md-1 pt-2 text-center">
                to
              </div>
              <div className="col-md-5">
                <input
                  type="date"
                  name="endDate"
                  className="form-control"
                  value={endDate}
                  onChange={(event) => this.handleOnChange(event)}
                />
              </div>
            </div>
          </div>
        </div>
        <div style={{ flex: '0 0 auto' }}>
          <button
            type="button"
            disabled={!this.enableSave()}
            className="btn btn-primary float-right"
            onClick={() => this.saveBroadcast()}
          >
            <i className="fa fa-save" />
            &nbsp;&nbsp;Save
          </button>
        </div>
      </div>
    )
  }

  enableSave() {
    const { subject, message, startDate, endDate } = this.state
    return subject && message && startDate && endDate
  }

  handleOnChange(event) {
    const { name, value } = event.target
    this.setState({ [name]: value })
  }

  saveBroadcast() {
    // To update config in ui.settings and reload it.
    this.props.saveConfigurationToServerAndReload('broadcast', this.state)
  }
}

const mapStateToProps = (state) => ({
  broadcast: state.configuration.ui.items.broadcast,
})

const mapDispatchToProps = (dispatch) => ({
  saveConfigurationToServerAndReload: (type, configuration) => dispatch(
    UiActions.saveConfigurationToServerAndReload(type, configuration)
  ),
})

export default connect(mapStateToProps, mapDispatchToProps)(Broadcast)
