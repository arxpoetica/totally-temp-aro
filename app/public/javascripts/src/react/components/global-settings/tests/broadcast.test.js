/* global test expect jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { Broadcast } from '../broadcast'
import { shallow } from 'enzyme'

const sampleConfiguration = { subject: '', body: '' }
// -----------------------------------------------------------------------------
test('Default component render', () => {
  const component = renderer.create(
    <Broadcast initialConfiguration={sampleConfiguration} />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('should call onChange prop with input value', () => {
  const onChangeMock = jest.fn()
  const component = shallow(<Broadcast
    handleSubjectChange={onChangeMock}
    handleBodyChange={onChangeMock} />)
  component.find('[type="text"]').simulate('change', { target: { value: 'Alert' } })
  component.find('textarea').simulate('change', { target: { value: 'Shutdown' } })

  expect(component.state('subject')).toEqual('Alert')
  expect(component.state('body')).toEqual('Shutdown')
})
