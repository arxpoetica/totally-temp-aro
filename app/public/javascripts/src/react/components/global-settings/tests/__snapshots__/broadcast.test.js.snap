// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default component render 1`] = `
<div
  className="no-collapse"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
    }
  }
>
  <div
    style={
      Object {
        "flex": "1 1 auto",
      }
    }
  >
    <div
      className="form-group"
    >
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        Disappear after 15 Seconds
      </label>
    </div>
    <div
      className="form-group"
    >
      <label>
        Subject
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Text
      </label>
      <textarea
        className="form-control"
        onChange={[Function]}
        rows="12"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "flex": "0 0 auto",
      }
    }
  >
    <button
      className="btn btn-primary float-right"
      onClick={[Function]}
    >
      <i
        className="fa fa-save"
      />
        Send
    </button>
  </div>
</div>
`;
